builders/build/canonical-contracts/precompiles/batch.md

```diff
@@ -0,0 +1,225 @@
+
+1. Expand the SimpleMessage contract under **Deployed Contracts**
+2. To the right of the **`messages`** button, insert `1`
+3. Press the blue **messages** button
+
+![SimpleMessage Confirmation](/images/builders/build/canonical-contracts/precompiles/batch/batch-7.png)
+
+The phrase "moonbeam" should appear underneath it. Congratulations! You have interacted with a function with the Batch precompile. 
+
+### Combining Subtransactions {: combining-subtransactions }
+
+So far, transferring native currency and interacting with functions have been separate, but they can be intertwined. 
+
+The following four strings can be combined as inputs for a batch transaction. They will transact 1 DEV to the public Gerald (`0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b`) account,  and interact with a predeployed SimpleMessage contract twice. Let's break it down:
+
+There are 3 subtransactions, so there are 3 addresses in the *to* input array. The first is the public Gerald account, the next two are a predeployed SimpleMessage contract. You can replace the last two with your own instance of SimpleMessage if you wish. Or, replace only one: you can interact with multiple contracts in a single message.
+
+```
+["0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b", "0xd14b70a55F6cBAc06d4FA49b99be0370D0e1BD39", "0xd14b70a55F6cBAc06d4FA49b99be0370D0e1BD39"]
+``` 
+
+There will also be 3 values for the *value* array. The first address in the *to* input array has to do with sending 1 DEV, so 1 DEV in Wei is within the array. The following two values are 0 because the function that their subtransactions are interacting with do not accept or require native currency.  
+
+```
+["1000000000000000000", "0", "0"]
+```
+
+We will need 3 values for the *call_data* array. Since transferring native currency does not require call data, the string is simply blank. The second and third values in the array correspond to invokations of **setMessage** that set messages to ids 5 and 6.
+
+```
+["", "0x648345c8000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000009796f752061726520610000000000000000000000000000000000000000000000", "0x648345c800000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000e61206d6f6f6e6265616d2070726f000000000000000000000000000000000000"]
+```
+
+The final input is for *gas_input*. This array will be left empty to forward all remaining gas to each subtransaction.
+
+```
+[]
+```
+
+Try sending a batched transaction with these inputs in Remix the same way [you did before](#function-interaction-via-precompile).
+
+And that's it! You've successfully interacted with the ERC-20 precompile using MetaMask and Remix!
+
+## Finding Call Data with Ethers.js {: #finding-call-data-with-ethers }
```

Comment:
You could generalize this to "Using Ethereum Dev Libaries" and used the tabbed table to present code snippets for web3.js, ethers and web3.py"

Check this for example: https://docs.moonbeam.network/builders/get-started/eth-compare/consensus-finality/#checking-tx-finality-with-ethereum-libraries